modules/videoio/src/cap_msmf.cpp

```diff
@@ -79,3840 +81,1706 @@
 #pragma comment(lib, "mfuuid")
 #pragma comment(lib, "Strmiids")
 #pragma comment(lib, "Mfreadwrite")
+#ifdef HAVE_DXVA
+#pragma comment(lib, "d3d11")
+#endif
 #if (WINVER >= 0x0602) // Available since Win 8
 #pragma comment(lib, "MinCore_Downlevel")
 #endif
 #endif
 
 #include <mferror.h>
 
-#ifdef WINRT
-    // for ComPtr usage
-#include <wrl/client.h>
-#ifdef __cplusplus_winrt
-#include <agile.h>
-#include <vccorlib.h>
-#endif
-
-#include <wrl\async.h>
-#include <wrl\implements.h>
-#include <wrl\module.h>
-#include <wrl\wrappers\corewrappers.h>
-#include <windows.media.capture.h>
-#include <windows.devices.enumeration.h>
-#ifdef HAVE_CONCURRENCY
-#include <concrt.h>
-#ifndef __cplusplus_winrt
-#include "wrl.h"
-#endif
-#include "ppltasks_winrt.hpp"
-#endif
-#else
 #include <comdef.h>
-#endif
 
 struct IMFMediaType;
-#ifndef WINRT
 struct IMFActivate;
 struct IMFMediaSource;
-#endif
 struct IMFAttributes;
 
 namespace
 {
 
-template <class T> void SafeRelease(T **ppT)
+#ifdef _DEBUG
+void DPOprintOut(const wchar_t *format, ...)
 {
-    if (*ppT)
+    int i = 0;
+    wchar_t *p = NULL;
+    va_list args;
+    va_start(args, format);
+    if (::IsDebuggerPresent())
+    {
+        WCHAR szMsg[512];
+        ::StringCchVPrintfW(szMsg, sizeof(szMsg) / sizeof(szMsg[0]), format, args);
+        ::OutputDebugStringW(szMsg);
+    }
+    else
     {
-        (*ppT)->Release();
-        *ppT = NULL;
+        if (wcscmp(format, L"%i"))
+        {
+            i = va_arg(args, int);
+        }
+        if (wcscmp(format, L"%s"))
+        {
+            p = va_arg(args, wchar_t *);
+        }
+        wprintf(format, i, p);
     }
+    va_end(args);
 }
+#define DebugPrintOut(...) DPOprintOut(__VA_ARGS__)
+#else
+#define DebugPrintOut(...) void()
+#endif
 
-#ifdef _DEBUG
-/// Class for printing info into console
-class DPO
+template <class T>
+class ComPtr
 {
 public:
-    ~DPO(void);
-    static DPO& getInstance();
-    void printOut(const wchar_t *format, ...);
-    void setVerbose(bool state);
-    bool verbose;
+    ComPtr() throw()
```

Comment:
throw() is [deprecated](http://en.cppreference.com/w/cpp/language/noexcept_spec).